api/models/User.js

```diff
@@ -240,4 +252,96 @@ UserSchema.statics.getUserPictureAboutCategories = function (userId) {
   return this.findById(userId, 'picture about categories').exec();
 };
 
+// Get top ranked user ids for a given investor category, descending
+// Ranking done by total reps
+// Only returns the username, picture, rank, and reps for the category
+UserSchema.statics.findTopRankedInvestors = function(category) {
+  return this.aggregate([
+    { $match: { "portfolio.category": category }},
+    { $unwind: "$portfolio" },
+    { $match: { "portfolio.category": category }},
+    { $sort: { "portfolio.rank": 1 }},
+    { $project: { _id: 1, picture: 1, "portfolio.rank": 1, username: 1 }},
+    { $limit: 10 },
+  ]).exec();
+};
+
+// Get top ranked user ids for a given expert category, descending
+// Ranking done by total reps
+// Only returns the username, picture, rank, and reps for the category
+UserSchema.statics.findTopRankedExperts = function(category) {
+  return this.aggregate([
+    { $match: { "categories.name": category }},
+    { $unwind: "$categories" },
+    { $match: { "categories.name": category }},
+    { $sort: { "categories.rank": 1 }},
+    { $project: { _id: 1, picture: 1, "categories.rank": 1, "categories.reps": 1, username: 1 }},
+    { $limit: 10 },
+  ]).exec();
+};
+
+// Get ranked user ids for a given expert category, descending
+// Ranking done by total reps
+// Only returns the IDs of the users in ranked order
+UserSchema.statics.findRankedExperts = function(category) {
+  return this.aggregate([
+    { $match: { "categories.name": category }},
+    { $unwind: "$categories" },
+    { $match: { "categories.name": category }},
+    { $sort: { "categories.reps": -1 }},
+    { $project: { _id: 1 }},
+  ]).exec();
+};
+
+// Get ranked user ids for a given expert category, decreasing
+// Ranking done by total dividends
+UserSchema.statics.findRankedInvestors = function(category) {
+  return this.aggregate([
+    { $match: { "portfolio.category": category }},
+    { $unwind: "$portfolio" },
+    { $unwind: "$portfolio.investments" },
+    { $match: { "portfolio.category": category } },
+    { $group: { _id: "$_id", dividends: { $sum: "$portfolio.investments.dividend" } }},
+    { $sort: { "dividends": -1 }},
+    { $project: { "_id": 1 } },
+  ]).exec();
+};
+
+// Update the rank for a given investor and category
+UserSchema.statics.updateRank = function(userId, categoryName, rank, expert, cb) {
+  if (expert) {
+    return this.update(
+      { _id: mongoose.Types.ObjectId(userId), "categories.name": categoryName },
+      { $set: { "categories.$.rank": rank } }
+    ).exec(cb);
+  } else {
+    return this.update(
+      { _id: mongoose.Types.ObjectId(userId), "portfolio.category": categoryName },
+      { $set: { "portfolio.$.rank": rank } }
+    ).exec(cb);
+  }
+};
+
+// Update the investments for a given investor and category
+// Used to update dividends upon a transaction occurring
+UserSchema.statics.updateInvestments = function(userId, categoryName, investments, cb) {
```

Comment:
When we create a transaction, we replace the entire investment list for the appropriate portfolio entry.